client/components/RestaurantList.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import RelatedRestaurants from './RelatedRestaurants';
+
+class RestaurantList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    };
+  }
+
+  render() {
+    const { data, cuisine, district } = this.props;
+    let list = [];
+    const type = data.filter(restaurant => restaurant.cuisine === cuisine);
+    const location = type.filter(restaurant => restaurant.district === district);
+    const listFilter = location;
+    if (listFilter.length === 6) {
+      list = listFilter;
+    } else if (listFilter.length > 6) {
+      list = listFilter.slice(0, 6);
+    } else if (listFilter.length < 6) {
+      const additionalRestaurants = data.filter(restaurant => restaurant.district === district);
+      for (let i = 0; i < additionalRestaurants.length; i += 1) {
+        const restaurant = additionalRestaurants[i];
+        if (!listFilter.includes(restaurant)) {
+          listFilter.push(restaurant);
+        }
+        if (listFilter.length > 5) {
+          break;
+        }
+      }
+      if (listFilter.length < 6) {
```

Comment:
I put that case there in case that there were not enough restaurants that matches the cuisine and district. In the case that we still had less than 6 restaurants, it will get random restaurants to fill the list to 6. 

I do agree that it is a rather repetitive process so I will review the logic for this part.